### PR TITLE
LINKTA-21 Lint rules

### PR DESCRIPTION
**Pull request by ChTiSh**
## Description
Fixes # (issue) LINKTA-21

## Type of change
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Additional Information:
Please see below for the brief explanations of all the lint rules currently included. 

General JavaScript/TypeScript Rules:
no-console: "warn" - Discourages using console.log statements, flagging them as warnings.
no-unused-vars: "error" - Ensures all declared variables are used, preventing clutter and potential confusion.
@typescript-eslint/consistent-type-imports: "error" - Enforces consistent usage of type-only imports for TypeScript.
@typescript-eslint/no-floating-promises: "error" - Prevents unhandled promise executions, reducing runtime errors.
no-shadow: "off" - Disables the rule that prevents variable declarations from shadowing outer declarations.
@typescript-eslint/no-shadow: "error" - TypeScript version of no-shadow, providing type-safe variable declarations.
prefer-rest-params: "error" - Encourages using rest parameters instead of the arguments object for better readability and maintainability.
@typescript-eslint/prefer-as-const: "error" - Prefers as const over explicit type definitions for literal expressions, enhancing type safety and inference.
React-Specific Rules:
react/destructuring-assignment: "off" - Allows props and state to be accessed directly, rather than through destructuring, for flexibility.
react/function-component-definition: "off" - No enforced component definition method, allowing both function declarations and arrow functions.
react/jsx-boolean-value: "off" - Permits JSX boolean props to be explicitly set, offering clarity in some cases.
react/jsx-filename-extension: "off" - Allows JSX syntax in files with any extension, not restricted to .jsx.
react/no-unused-prop-types: "warn" - Warns about defined prop types that are not used in components, aiding in code cleanliness.
react/prop-types: "off" - Disables PropTypes validation, likely relying on TypeScript for props validation instead.
react-hooks/rules-of-hooks: "error" - Ensures Hooks are called in the correct order and scope, maintaining the rules of Hooks.
react-hooks/exhaustive-deps: "warn" - Warns if dependencies in Hooks like useEffect are not properly specified, preventing unintended side effects.

Prettier and Style Rules:
prettier/prettier: "error" - Enforces Prettier formatting rules, ensuring code style consistency.
quotes: "error" - Enforces single quotes for strings, with allowances for template literals and avoiding escape.
Import/Export and Other Rules:
no-relative-import-paths/no-relative-import-paths: "warn" - Warns against using relative paths for imports, preferring absolute paths or aliases.
arrow-body-style: "off" - No enforced style for arrow function bodies, allowing both block and concise bodies.
prefer-arrow-callback: "off" - Allows any function type for callbacks, not enforcing arrow functions.
import/no-default-export: "off" - No restriction on default exports, allowing flexibility in module design.
import/no-extraneous-dependencies: "off" - No enforcement on importing only from dependencies in package.json, accommodating varied project structures.
no-nested-ternary: "off" - Permits nested ternary operators, which can be concise but potentially less readable.
no-plusplus: "off" - Allows unary increment and decrement operators (++, --), which can be concise but potentially confusing.
no-prototype-builtins: "off" - Permits direct calls to Object.prototype methods, which can be clearer but may have edge cases.

Configuration Meta:
env: Specifies environments (browser, ES2021, Node.js), providing global variables.
parser: Uses @typescript-eslint/parser for parsing TypeScript.
parserOptions: Configures ECMAScript version and module system.
plugins: Includes essential plugins for TypeScript, React, Prettier, and accessibility.
root: Marks this as the root configuration, stopping ESLint from looking further up.
ignorePatterns: Instructs ESLint to ignore specified directories.
overrides: Provides specific rules for .js files, accommodating mixed TypeScript/JavaScript codebases.
extends: Inherits rules from recommended sets, ensuring a solid base for linting and formatting.

### CODE REVIEW

**Review by LucieSeidler — APPROVED** [review]
we will still need to add remaining packages and scripts within package.json

**Comment by kinskiwu** [comment]
Please investigate and address the following eslint config issues before merge:
- The following rules appear to be custom rules or require plugins. Please either remove them or add the relevant plugin (consider adding eslint-plugin-react-hooks to package.json and add “react-hooks” to the plugins array (e.g. "plugins": ["react-hooks”]). 
  - "no-relative-import-paths/no-relative-import-paths" 
  - "react-hooks/rules-of-hooks"
  - "react-hooks/exhaustive-deps"
- To use ‘@typescript-eslint/no-floating-promises’, we will need to include a parserOptions object that has a project property pointing to our tsconfig.json file (e.g. "parserOptions": {"project": "./tsconfig.json"})

Note: may be a good idea to merge this PR after we have an agreement on the tech stack and update the package.json file and the remaining config files

**Comment by ChTiSh** [comment]
Hi @kinskiwu, I just adjusted the plugins and parseOptions.

File: .prettierrc.json
Comment (dcpavel) [inline .prettierrc.json:1]: I’m not on the front end, but I find singleAttributePerLine: true to be a bit more readable.

**Review by dcpavel — CHANGES_REQUESTED** [review]
I had a couple of questions about the eslint choices before I agree to the PR.

File: .eslintrc.json
Comment (ChTiSh) [reply .eslintrc.json:1]: Good call, I will try to get it more organized and grouped.
Regarding the "es2021" and ecmaVersion of "latest", it was default setting suggested by ESlint. I do see that eslint's default parser Espree is now supporting es2024 as well. I can change the env to es2024 actually to make it more consistent. 
I did a bit of research about would setting ecmaVersion to latest would or not conflict with different env setting, TLDR, env-global variables, parseOptions-syntax, doesn't conflict, but here is a post about it:
https://www.reddit.com/r/learnjavascript/comments/16q0mwi/eslint_configuration_help/

I will add the ecmaFeatures for jsx, thank you so much for calling it out. 

For "rootDir", I actually prefer to leave it as is. 
### `rootDir`

Useful when auto-fixing and the rootDir should not be included in the absolute path.

Examples of code for this rule:

```js
// when not configured:
import Something from "../../components/something";

// will result in
import Something from "src/components/something";
```

```js
// when configured as { "rootDir": "src" }
import Something from "../../components/something";

// will result in
import Something from "components/something";
//                     ^- no 'src/' prefix is added
```

File: .prettierrc.json
Comment (ChTiSh) [reply .prettierrc.json:1]: Haha, I can see this could be a huge debate. 
I will change to true for now, and see what happens.

**Review by dcpavel — APPROVED** [review]

**Pull request merged**
